pkg/mapper/dynamicfile/mapper.go

```diff
@@ -29,14 +30,14 @@ func (m *DynamicFileMapper) Start(stopCh <-chan struct{}) error {
 	return nil
 }
 
-func (m *DynamicFileMapper) Map(canonicalARN string) (*config.IdentityMapping, error) {
-	canonicalARN = strings.ToLower(canonicalARN)
+func (m *DynamicFileMapper) Map(identity *token.Identity) (*config.IdentityMapping, error) {
+	canonicalARN := strings.ToLower(identity.CanonicalARN)
 
-	rm, err := m.RoleMapping(canonicalARN)
+	rm, err := m.RoleMapping(identity.UserID)
```

Comment:
Can we support mapping by UserId and ARN? For example: if only UserId is set, use it.  If only ARN is set, use it.  If both are set, use UserId and compare ARN to ensure a match.

Reply:
dynamic file is new mode and we have decided to use aws UserId because it is global unique. assuming we do not go back to IdentityARN any more since we already know the weakness of IdentityARN for non-exist role and recreated role?

Reply:
the new requirement is to replace the IdentityArn with PrincipleId because IdentityArn has weakness in non-exist role and recreated-role, is it?

Reply:
Yes, but its also a decision that we can leave up to the user, and we can do this in a backwards compatible way for other backend modes.